jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/bucketEditor/S3TreeTable.kt

```diff
@@ -65,4 +85,51 @@ open class S3TreeTable(private val treeTableModel: S3TreeTableModel) : TreeTable
     fun invalidateLevel(node: S3TreeNode) {
         node.parent?.removeAllChildren()
     }
+
+    private fun createDropTarget(): DropTarget {
+        val dropTarget = DropTarget()
+        try {
+            dropTarget.addDropTargetListener(object : DropTargetAdapter() {
+                override fun drop(dropEvent: DropTargetDropEvent) {
+                    try {
+                        dropEvent.acceptDrop(DnDConstants.ACTION_COPY_OR_MOVE)
+                        val data = dropEvent.transferable.getTransferData(DataFlavor.javaFileListFlavor) as List<File>
+                        val row = rowAtPoint(dropEvent.location).takeIf { it >= 0 } ?: return
+                        val node = getNodeForRow(row) ?: return
+
+                        val lfs = LocalFileSystem.getInstance()
+                        val virtualFiles = data.mapNotNull {
+                            lfs.findFileByIoFile(it)
+                        }
+
+                        ApplicationManager.getApplication().executeOnPooledThread {
+                            val action = UploadObjectAction(bucketVirtual, this@S3TreeTable)
+                            virtualFiles.forEach {
+                                try {
+                                    action.uploadObjectAction(s3Client, project, it, node)
+                                } catch (e: Exception) {
+                                    LOG.error("unable to upload to s3 from drag and drop", e)
```

Comment:
rather than `LOG.error` which surfaces as an uncaught exception in the IDE we should `use the `Exception.notify` extension function in `NotificationUtils` (which will also log the error as a `warn`).

Reply:
makes sense, changed